onnxruntime/test/onnx/tensorprotoutils.cc

```diff
@@ -22,16 +23,6 @@ struct OrtStatus {
 
 namespace onnxruntime {
 namespace test {
-#ifdef __GNUC__
-constexpr inline bool IsLittleEndianOrder() noexcept { return __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__; }
```

Comment:
This is a compiler time constant, so compiler can do dead code elimination based on that.  In the else branch, we may just always return true.

Reply:
True, however the other version with reinterpret_cast is not a compile time constant. I think comparison of two constant enumeration values should also be known at compile time, right?

Reply:
This is correct.

---
In reply to: [336668041](https://github.com/microsoft/onnxruntime/pull/2181#discussion_r336668041) [](ancestors = 336668041)